tap_counter/ViewController.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  ViewController.swift
+//  tap_counter
+//
+//  Created by Michail on 9/10/16.
+//  Copyright © 2016 mishaparadising. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+    
+    var counter = 0;
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view, typically from a nib.
+        
+        tapCounter.text = String(counter);
+        
+        self.navigationItem.hidesBackButton = true;
+        
+        let resetButton = UIBarButtonItem(title: "Reset", style: UIBarButtonItemStyle.Bordered, target: self, action: Selector("resetButton"))
+        self.navigationItem.leftBarButtonItem = resetButton;
```

Comment:
Do not use `self` in synchronous code, only async block should refer to `self` because they work that way.